src/time.cpp

```diff
@@ -69,6 +71,27 @@ namespace g3 {
    /// * format string must conform to std::put_time
    /// This is similar to std::put_time(std::localtime(std::time_t*), time_format.c_str());
 
+   std::string localtime_formatted(const timespec &time_snapshot, const std::string &time_format) {
+      ushort fcount = 0;
+      for(auto found_ptr = time_format.c_str(); (found_ptr = strstr(found_ptr, "%f")) != nullptr; found_ptr++, fcount++);
+      const size_t format_size = time_format.length() + fcount * 7 + 1;  // len(format) + num(nanosec size - len("%f")) + zero
```

Comment:
I will pull down your changes and try them.

What does the "f" stand for?  Ref: http://en.cppreference.com/w/cpp/io/manip/put_time

since you  are expanding the API for the format string you can make it obvious (more so than the std)
Why not spelling it out? 
%NS 
Or 
 %NanoSec that way it is ensured to avoid API clashes